### PR TITLE
cpu: centralized startup code for cortexm CPUs

### PR DESCRIPTION
**Pull request by haukepetersen**
The startup code and exception handlers for cortexm based CPUs were pretty much identical, so centralizing them makes much sense.

@gebart, @jfischer-phytec-iot: I was not blunt enough to touch the kinetis code (as it still differs quite a bit from how we implemented all other platforms). Would you care to have a look at this as a follow-up?

Note: github does a shitty job on showing the actual diffs here. The changes are not as bad as it seems...

### CODE REVIEW

File: cpu/cortexm_common/include/vectors_cortexm.h [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+#ifndef VECTORS_DEFAULT_H_
+#define VECTORS_DEFAULT_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Use this macro to make interrupt functions over writable with the
```
Comment (gebart) [inline cpu/cortexm_common/include/vectors_cortexm.h]: s/over writable/overridable/

File: cpu/cortexm_common/include/vectors_cortexm.h [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+ *          dummy_hanler as fallback in case they are not implemented
```
Comment (gebart) [inline cpu/cortexm_common/include/vectors_cortexm.h]: dummy_handler

**Comment by gebart** [comment]
@haukepetersen I will try to port this to k60, and possibly kw2x during next week, if time permits.

**Comment by haukepetersen** [comment]
fixed spelling

**Comment by jfischer-phytec-iot** [comment]
@haukepetersen @gebart I was so free and have adapted it to kinetis. @haukepetersen Can I open a PR against your Branch?

**Comment by jfischer-phytec-iot** [comment]
@haukepetersen I was impatient and did it https://github.com/haukepetersen/RIOT/pull/11. @gebart Can you review and test it on mull it please?

**Comment by haukepetersen** [comment]
@jfischer-phytec-iot: looking great, merged!

**Comment by gebart** [comment]
I get a hard fault at the end of pre_startup on k60. The return pointer gets overwritten by the canary value and popped. The original implementation inlined the call so that there was no `pop $pc`

**Comment by gebart** [comment]
You could delete the stack canary filling, or move it to Cortex M common. 

The purpose of it is to fill the ISR stack with a known value so that we can measure ISR stack usage by looking through the memory.

**Comment by jfischer-phytec-iot** [comment]
Damn, I miss it, of course it raises hardfault.

**Comment by jfischer-phytec-iot** [comment]
@gebart @haukepetersen corrected: https://github.com/haukepetersen/RIOT/pull/12

**Comment by gebart** [comment]
I'm on mobile so I can't comment inline. Looks good, I don't think you need
to keep the estack extern in the k60 file though.

**Comment by haukepetersen** [comment]
@jfischer-phytec-iot: I merged your fix without really looking into it too much. I will look over this complete PR in detail once more on Monday.

**Comment by haukepetersen** [comment]
Ok, as far as I see it, the changes look good and this PR should be ready for merge. Should I squash and start Travis?

**Comment by jfischer-phytec-iot** [comment]
@haukepetersen do it :-)

**Comment by haukepetersen** [comment]
did, ready for Travis :-)

**Comment by gebart** [comment]
```
cpu/cortexm_common/vectors_cortexm.c:74: trailing whitespace.
ERROR: This change introduces new whitespace errors
```

**Comment by haukepetersen** [comment]
fixed whitespace issue and rebased.

**Comment by jfischer-phytec-iot** [comment]
I guess we can merge it ...

**Comment by haukepetersen** [comment]
just needs a little rebase as #3095 was finally merged. Doing this now.

**Comment by haukepetersen** [comment]
rebased, hopefully I didn't break anything... So lets wait for Travis again and merge once he is happy?

**Comment by jfischer-phytec-iot** [comment]
Travis happy, go

**Comment by haukepetersen** [comment]
yeah!

**Pull request merged**